Creating example kontainer engine driver

nathan-jenan-rancher opened the pull request:
This is an example kontainer engine driver. It is basically just a copy of the
existing google kuberentes engine driver, but serves as an example of how to
create a driver and package it for use with Rancher.

Issue:
https://github.com/rancher/rancher/issues/12698

nathan-jenan-rancher committed:
Update vendor

StrongMonkey reviewed (commented):
Don't commit binary into the repo.

nathan-jenan-rancher committed:
Creating example kontainer engine driver
This is an example kontainer engine driver. It is basically just a copy of the
existing google kuberentes engine driver, but serves as an example of how to
create a driver and package it for use with Rancher.

Issue:
rancher/rancher#12698

StrongMonkey commented:
@nathan-jenan-rancher is your kontainer-engine actually provisioning real cluster or just provide fake call(like create schema)?

StrongMonkey commented:
If it is just providing the call to create schema i would rather delete all other code is not used by tests to make it a little bit clean. But it depends on you.

StrongMonkey commented:
LGTM

liuxiaopiao commented:
Thanks for the example. So we can follow this example to add new Kubernetes cloud provider.  Is there any more detailed guide for user to follow which other rancher repo we need to  contribute if we want to add one new Kubernetes  cloud provider for Rancher? Thanks a lot
